src/hotspot/share/oops/method.cpp

```diff
@@ -144,8 +144,6 @@ void Method::release_C_heap_structures() {
 #if INCLUDE_JVMCI
     FailedSpeculation::free_failed_speculations(method_data()->get_failed_speculations_address());
 #endif
-    // Destroy MethodData
-    method_data()->~MethodData();
```

Comment:
You still need this here.  This is for the class unloading path.

Reply:
I agree this needs to remain else we are removing other destructor calls, not just adding back the one currently missing. But it raises the question for me as to why we have two distinct code paths to clean up the MethodData: one from `Method::release_C_heap_structures` and the other from `Method::deallocate_contents`? Aren't there code paths where both of those would be executed and we would then call the destructor twice?

Reply:
I just saw the other comments point to the comment in JDK-8298346 so no need to respond to this.